File: remove-linked-list-elements/remove-linked-list-elements.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        ListNode current = head;
        ListNode temp = null;
        while(current!=null){
            if(current.val==val){
                if(current==head){
                    head=head.next;
                    current=head;
                    continue;
                }else{
                    temp.next=current.next;
                    current=temp.next;
                    continue;
                }
            }
            temp=current;
            current=current.next;
        }
        return head;
    }
}

File: 203-remove-linked-list-elements/remove-linked-list-elements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        # #Approach1
        # if head == None or (head.next == None and head.val != val):
        #     return head
        # ans = None
        # previous = None
        # current = head

        # while(current != None):
        #     if current.val != val and ans == None:
        #         ans = current 
        #         previous = current 
        #     elif current.val != val:
        #         previous.next = current
        #         previous = current 
        #     current = current.next
        # if previous != None:
        #     previous.next = None
        # return ans

        #Approach 2
        holder = ListNode()
        ans = holder
        runner = head

        while(runner != None):
            if runner.val != val:
                holder.next = runner
                holder = holder.next
            runner = runner.next
        holder.next = None
        return ans.next